### PR TITLE
ENT-2158 | Added XAPILearnerDataTransmissionAudit model and more logg…

### PR DESCRIPTION
**Pull request by hasnain-naveed**
…ing.

**Description:** Describe in a couple of sentence what this PR adds

**JIRA:** Link to JIRA ticket

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Merge checklist:**

- [ ] New requirements are in the right place
    - `base.in` if needed in production, but edx-platform doesn't install it.
    - `test-master.in` if edx-platform pins it, with a matching version.
- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] Called `make static` for webpack bundling if any static content was updated.
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)
- [ ] edx-platform PR (be sure to include edx-platform requirements upgrades that were present in this PR)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: integrated_channels/xapi/models.py [hunk 1]
```diff
@@ -65,3 +66,51 @@ def authorization_header(self):
...
+
+    user = models.ForeignKey(
+        User,
+        blank=False,
+        null=False,
+        related_name='xapi_transmission_audit',
+        on_delete=models.CASCADE,
+    )
+    enterprise_course_enrollment_id = models.PositiveIntegerField(db_index=True, blank=True, null=True)
+    course_id = models.CharField(max_length=255, blank=False, null=False)
```
Comment (ziafazal) [inline integrated_channels/xapi/models.py]: need an index(`db_index=True`) on `course_id` field.

File: integrated_channels/xapi/models.py [hunk 2]
```diff
@@ -65,3 +66,51 @@ def authorization_header(self):
...
+            '<XAPILearnerDataTransmissionAudit {transmission_id} for enterprise enrollment '
+            '{enterprise_course_enrollment_id}, XAPI user {user_id}, and course {course_id}>'.format(
+                transmission_id=self.id,
+                enterprise_course_enrollment_id=self.enterprise_course_enrollment_id,
+                user_id=self.user.id,
+                course_id=self.course_id
+            )
+        )
+
+    def __repr__(self):
```
Comment (ziafazal) [inline integrated_channels/xapi/models.py]: do we really need this `__repr__` method here?

File: integrated_channels/xapi/utils.py [hunk 3]
```diff
@@ -25,7 +31,12 @@ def send_course_enrollment_statement(lrs_configuration, course_enrollment):
         course_enrollment.course,
         context={'site': lrs_configuration.enterprise_customer.site}
     )
-
+    LOGGER.info(
+        "Sending course enrollment for user {username} for course {course_key}".format(
```
Comment (ziafazal) [inline integrated_channels/xapi/utils.py]: nit: Message should mention xAPI. something like this 
```
Sending course enrollment to xAPI for user...
```

File: integrated_channels/xapi/utils.py [hunk 4]
```diff
@@ -50,7 +61,13 @@ def send_course_completion_statement(lrs_configuration, user, course_overview, c
         course_overview,
         context={'site': lrs_configuration.enterprise_customer.site}
     )
-
+    LOGGER.info(
+        "Sending course completion for user '{username}' for course '{course_key}' with {percentage}%".format(
```
Comment (ziafazal) [inline integrated_channels/xapi/utils.py]: nit: Message should mention xAPI. something like this 
```
Sending course completion to xAPI for user...
```

File: tests/test_integrated_channels/test_xapi/test_commands/test_send_course_completions.py [hunk 5]
```diff
@@ -234,17 +249,30 @@ def test_command(self, mock_send_course_completion_statement):
...
 
-        assert mock_send_course_completion_statement.call_count == 5
+        assert mock_send_completion_statement.call_count == 5
+
+    def test_get_learner_transmission_object(self):
+        """
+        Make sure get_learner_transmission_object method works as expected.
+        """
+        # Import is placed here because if placed at the top it affects mocking.
+        from integrated_channels.xapi.management.commands.send_course_completions import Command
```
Comment (ziafazal) [inline tests/test_integrated_channels/test_xapi/test_commands/test_send_course_completions.py]: I'm not able to understand the reason for local import here.

File: tests/test_integrated_channels/test_xapi/test_models.py [hunk 6]
```diff
@@ -43,3 +43,30 @@ def test_authorization_header(self):
...
+        Test the string representation of the model.
+        """
+        expected_string = '<XAPILearnerDataTransmissionAudit {id} for enterprise enrollment {ece_id}, XAPI user ' \
+                          '{user_id}, and course {course_id}>'.format(
+                              id=self.xapi_learner_transmission.id,
+                              ece_id=self.xapi_learner_transmission.enterprise_course_enrollment_id,
+                              user_id=self.xapi_learner_transmission.user_id,
+                              course_id=self.xapi_learner_transmission.course_id
+                          )
+        assert expected_string == self.xapi_learner_transmission.__repr__()
```
Comment (ziafazal) [inline tests/test_integrated_channels/test_xapi/test_models.py]: nit: `str(self.xapi_learner_transmission)`

File: integrated_channels/xapi/management/commands/send_course_enrollments.py [hunk 7]
```diff
@@ -118,15 +120,42 @@ def send_xapi_statements(self, lrs_configuration, days):
...
+            else:
+                LOGGER.info(
+                    "Successfully send the course enrollment to xAPI for user '{username}' for course "
+                    "'{course_id}'".format(
+                        username=course_enrollment.user.username,
+                        course_id=course_id
                     )
                 )
+                status = 200
+            XAPILearnerDataTransmissionAudit.objects.get_or_create(
```
Comment (ziafazal) [inline integrated_channels/xapi/management/commands/send_course_enrollments.py]: nit: add user keyworad
```
 XAPILearnerDataTransmissionAudit.objects.get_or_create(
  user=course_enrollment.user,
   course_id=course_id,
..........

```

File: integrated_channels/xapi/management/commands/send_course_completions.py [hunk 8]
```diff
@@ -133,18 +133,40 @@ def send_xapi_statements(self, lrs_configuration, days):
...
+            error_message = None
+            user = users.get(persistent_course_grade.user_id)
+            course_overview = course_overviews.get(persistent_course_grade.course_id)
+            course_grade = CourseGradeFactory().read(user, course_key=persistent_course_grade.course_id)
+            xapi_transmission_queryset = XAPILearnerDataTransmissionAudit.objects.filter(
+                user=user,
+                course_id=persistent_course_grade.course_id
+            )
+            if not xapi_transmission_queryset.exists():
+                continue
```
Comment (ziafazal) [inline integrated_channels/xapi/management/commands/send_course_completions.py]: Before `continue` we need to log a message with info.

**Comment by georgebabey** [comment]
Looks good - just needs version bump and changelog

**Pull request merged**